src/constructor/load.py

```diff
@@ -0,0 +1,221 @@
+
+    Returns:
+        state_dict_with_prefix: Prefixed state dict.
+    """
+    state_dict_with_prefix = OrderedDict()
+    # remove spaces and dots
+    prefix = prefix.strip(' .')
+    for key, value in state_dict.items():
+        if not key.startswith(prefix):
+            key = prefix + '.' + key
+        state_dict_with_prefix[key] = value
+    return state_dict_with_prefix
+
+
+def get_straighten_keys(require_key: str, model_keys: List[str]) -> List[str]:
+    """Create straighten model keys from require.
+
+    Args:
+        require_key: The key to being straightened.
+        model_keys: The model keys.
+
+    Returns:
+        straighten_keys: Straightened keys.
+    """
+    straightened_keys = []
+    for key in model_keys:
+        if key.startswith(require_key):
+            straightened_keys.append(key)
+    return straightened_keys
+
+
+def generate_required_state_dict(base_state_dict: OrderedDict[str, torch.Tensor], 
+                                 overridden_name2state_dict: Dict[str, OrderedDict[str, torch.Tensor]], 
+                                 exclude_keys: List[str],
+                                 model_keys: List[str]) -> OrderedDict[str, torch.Tensor]:
+    """Generate state dict, which should be loaded from 4 main components: base state dict, overridden state dicts,
+    exclude keys and model_keys.
+
+    Base state dict values are overridden with overridden_name2state_dict, where the key in overridden_name2state_dict - 
+    module name which should be overridden.
+    If overridden_name2state_dict has many state dicts for one key in the base state dict, then the state dict is 
+    selected whose overridden_name module is closer to the key i.e. choose the deepest overridden_name, where depth 
+    is the number of dots.
+    After the base state dict had been overridden, it is necessary to remove keys whose names begin with any
+    key in exclude_keys.
+
+    Args:
+        base_state_dict: Base state dict that should be loaded.
+        overridden_name2state_dict: Dicts of module key to state dict, which should override base state dict.
+        exclude_keys: Module keys that should not be loaded.
+        model_keys: Model state dict keys.
+
+    Returns:
+        required_state_dict: State dict obtained by override base state dict by overridden state dict, which not 
+            contain the keys starting with exclude keys.
+
+    Raises:
+        ValueError: If base_state_dict, overridden_name2state_dict or exclude_keys not match with the model keys.
+    """
+    required_state_dict = OrderedDict()
+
+    # Check base state dict
+    check_state_dict_keys(base_state_dict, model_keys)
+
+    # Add prefix for every overridden state dict
+    overridden_full_name2state_dict = dict()
+    for overridden_name, state_dict in overridden_name2state_dict.items():
+        prefixed_state_dict = get_state_dict_with_prefix(prefix=overridden_name, state_dict=state_dict)
```

Comment:
I suggest leaving the job of putting the correct prefixes to a user because it's unclear where they start